verilog/tools/preprocessor/verilog_preprocessor.cc

```diff
@@ -42,25 +51,177 @@ static absl::Status StripComments(const SubcommandArgsRange& args,
+  verible::TokenSequence lexed_sequence;
+  for (lexer.DoNextToken(); !lexer.GetLastToken().isEOF();
+       lexer.DoNextToken()) {
+    // For now we will store the syntax tree tokens only, ignoring all the
+    // white-space characters. however that should be stored to output the
+    // source code just like it was, but with conditionals filtered.
+    if (verilog::VerilogLexer::KeepSyntaxTreeTokens(lexer.GetLastToken()))
+      lexed_sequence.push_back(lexer.GetLastToken());
+  }
+  verible::TokenStreamView lexed_streamview;
+  // Initializing the lexed token stream view.
+  InitTokenStreamView(lexed_sequence, &lexed_streamview);
+  verilog::VerilogPreprocessData preprocessed_data =
+      preprocessor.ScanStream(lexed_streamview);
+  auto& preprocessed_stream = preprocessed_data.preprocessed_token_stream;
+  for (auto u : preprocessed_stream) outs << *u << '\n';
+  // for debugging.
+  for (auto& u : preprocessed_data.errors) outs << u.error_message << '\n';
+
+  //  parsing just as a trial
+  std::string post_preproc;
+  for (auto u : preprocessed_stream) post_preproc += std::string{u->text()};
+  std::string source_view{post_preproc};
+  verilog::VerilogAnalyzer analyzer(source_view, "file1", config);
+  auto analyze_status = analyzer.Analyze();
+  /* const auto& mydata = analyzer.Data().Contents(); */
+  /* outs<<mydata; */
+
+  /* TODO(karimtera): regarding conditionals
+   1) Modify VerilogPreprocess config to have a configuration to generate SV
+   source codes for all possible variants. 2) Then use parser, directly from
+   VerilogAnalyzer or from VerilogParser to have less dependences. 3) Now, we
+   should have multiple trees, we need to merge them as described by Tom in
+   Verible's issue. 4) Finally, travese the tree and output the chosen path
+   based on definitions.
+  */
+  return absl::OkStatus();
+}
+
+static absl::Status MultipleCU(const SubcommandArgsRange& args, std::istream&,
+                               std::ostream& outs, std::ostream&) {
+  if (args.empty()) {
+    return absl::InvalidArgumentError("Missing file arguments.");
+  }
+  for (const char* source_file : args) {
+    outs << source_file << ":\n";
+    auto status = PreprocessSingleFile(source_file, std::cin, outs, std::cerr);
+    if (!status.ok()) return status;
+    outs << '\n';
+  }
+  return absl::OkStatus();
+}
+
+static absl::Status GenerateVariants(const SubcommandArgsRange& args,
+                                     std::istream&, std::ostream& outs,
+                                     std::ostream&) {
+  const int limit_variants = absl::GetFlag(FLAGS_limit_variants);
+  if (args.size() > 1) {
+    return absl::InvalidArgumentError(
+        "ERROR: generate-variants only works on one file.");
+  }
+  const char* source_file = args[0];
+  std::string source_contents;
+  if (auto status = verible::file::GetContents(source_file, &source_contents);
+      !status.ok()) {
+    std::cerr << "ERROR: passed file can't be open\n.";
```

Comment:
Similar here as in the preprocess function; this should probably go to the message stream and include filename and status message.